### PR TITLE
Feat/layer validation

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

* Adds layer threshold format during API data validation
* Corrects how the seconds until next reset is calculated for the healthcheck
* Merges main to bring the audit changes to the staging branch

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: audit/Contribution_Service_Diagram_Attacks.drawio [hunk 1]
```diff
@@ -0,0 +1 @@
...AzxVa1dSK0GY6eThoB418HrKB8HimRDDK2sBpsyyV/9SMMignod/evrr7bZClbjYYmT3O3PcO9TS/hRVvzn++OlrYsGFIgqd+dbTeQhdVFaw3SQZVoYyg4fMFVfSpDKchqFn1u31QB+0CbqOsPrX6a/HXm9bQx5VY1MuO8zWLaxCZYUi8cnPWi9cYLxjSjCJrwiZGS4ToH5XWl8kuIgz+O4pJ/MKUWhtcYXJFoEQZtV0kl9GZbejG0W19zWXUcOyCjKqGWZZRWVlA7SxtpTfpAeu3Jv9IOWqavR3ZuzL0cvb24xKvIDddjCDVItJvfewoOM0lwTKtoiQ4ZUnotT5myHqAWvJA/mFB2NxHHMktDAJEn68XYor1aM2WaDijRy/E0LoDlkXXc0xWsXdJmOFBPJ8HXS510S+EsSVJ8fSLcYdZ2tsNc1E4wiHaZqHhMwb3TNwgAoaODY0OU7Of1h6kMbppyM+MvecTZvqeMcSa0GZmD518N2RvDmqxyBKVw5DIxQRJxp8v6rrQXCPzuHqGvezxdg37iseTXpYM+qXxB30q9vcX18btzw8O+XNmza2t/ubT2V/S91hVxDjzAN2nbX8XuQ5AN4BR5LuTI02AbHZKekdG4atz8yJOlh+2KgGteauprGIzkUWVtbGslEJAjmg/bX2q0NZsndrWLL4nQu+urbSaigU++xvRLSbf2GAAd3zrqQIpZf57lGiVslo4Y6Oeh5M0MvtnBiP06cPbiolcsShL/ytb0Q8X8klLStfsafgvPvvG/3hzRMk0fkFRSR9IGK/a9xMcA0srqwhDFpVbLUTlUujKjsF5DFSGwybiYleJmQlXnErhLOK/AhwXdzCJqYwDXg1Q1sT/wVxA9ukb4qR4kUTof8hRky5xfvs2uVxiT8YKtaEW8y+WpFrQL9QNeoOGsQYgbw2miuYMaREaNxyZg5oOoEzZ3/2PWuQ18RVTjfvDhU2fenf94VLiy/zVEbCooTujY9GhOM907GnW6pCFt4o2VcBpHGcJRdUO30wlpWW5OJNYD8XnhsdNmmhcGKf5hzYpprCrQLclAUZX/aZS6jV5p1J/AttUxIYTHWEDgVhzbCo04kJmz2pa8hqjKof/6Ths4rtRJG/a7ddfk/TO/xauSuEyxDcln+oDlRZq7T117PDwf8Ik0w//sw64+h8=</diagram></mxfile>
```
Comment (dvilelaf) [inline audit/Contribution_Service_Diagram_Attacks.drawio:1]: Audit was merged into main, so I've merged main into staging to bring those latest changes

File: packages/valory/skills/dynamic_nft_abci/behaviours.py [hunk 2]
```diff
@@ -339,9 +343,14 @@ def validate_api_data(self, data: Dict) -> bool:
...
-                for threshold_data in i["values"]:
+                for layer_data in i["values"]:
+
+                    # Check the layer and threshold format
+                    if not all(re.match(THRESHOLD_REGEX, t) for t in layer_data):
+                        return False
+
                     thresholds = [
-                        int(img_data.split(":")[0]) for img_data in threshold_data
+                        int(img_data.split(":")[0]) for img_data in layer_data
```
Comment (dvilelaf) [inline packages/valory/skills/dynamic_nft_abci/behaviours.py:353]: We check that the layer threshold data follows the format `score:image_hash`

File: packages/valory/skills/dynamic_nft_abci/handlers.py [hunk 3]
```diff
@@ -66,6 +66,7 @@
 OK_CODE = 200
 NOT_FOUND_CODE = 404
 BAD_REQUEST_CODE = 400
+AVERAGE_PERIOD_SECONDS = 10
```
Comment (dvilelaf) [inline packages/valory/skills/dynamic_nft_abci/handlers.py:69]: I've run +100 periods to get this value. This number represents how much it takes for the agents to complete a full period (not accounting for the reset and pause time), and it is rounded up a bit.

File: packages/valory/skills/dynamic_nft_abci/handlers.py [hunk 4]
```diff
@@ -290,7 +291,7 @@ def _handle_get_health(
 
             seconds_since_last_reset = current_time - last_update_time
             seconds_until_next_update = (
-                observation_interval - seconds_since_last_reset
+                AVERAGE_PERIOD_SECONDS + observation_interval - seconds_since_last_reset
```
Comment (dvilelaf) [inline packages/valory/skills/dynamic_nft_abci/handlers.py:294]: We were not taking into account that observation_interval is the wait time (imo we should rename it). So to estimate when the next update will happen, we need both the average period time + the wait time.

**Review by 0xArdi — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**